cmd/prometheus/main.go

```diff
@@ -227,13 +227,15 @@ func reloadConfig(filename string, rls ...Reloadable) (err error) {
 		return fmt.Errorf("couldn't load configuration (-config.file=%s): %v", filename, err)
 	}
 
-	// Apply all configs and return the first error if there were any.
+	failed := false
 	for _, rl := range rls {
-		if err != nil {
-			err = rl.ApplyConfig(conf)
-		} else {
-			rl.ApplyConfig(conf)
+		if err := rl.ApplyConfig(conf); err != nil {
+			log.Error("Failed to apply configuration: ", err)
```

Comment:
Hmm, wouldn't that also confuse people, because the initial config load is also just a reload internally? A startup config failure is probably the most common for users with a bad config. The question is whether it's worth differentiating the two cases in the log message. The caller already differentiates in the final log message (https://github.com/prometheus/prometheus/blob/master/cmd/prometheus/main.go#L120 vs. https://github.com/prometheus/prometheus/blob/master/cmd/prometheus/main.go#L134-L144), so maybe that's enough?